src/main/java/com/minecolonies/api/util/Pond.java

```diff
@@ -16,165 +16,43 @@ public final class Pond
+     * @param water The coordinate to check.
      * @return true if water.
      */
-    public static boolean checkWater(@NotNull final BlockGetter world, @NotNull final BlockPos water, final WaterPathResult result)
+    public static boolean checkWater(@NotNull final BlockGetter world, @NotNull final BlockPos water)
     {
-        if (checkWater(world, water, WATER_POOL_WIDTH_REQUIREMENT, WATER_POOL_LENGTH_REQUIREMENT))
-        {
-            result.pond = water;
-            return true;
-        }
-        return false;
-    }
+        final BlockPos.MutableBlockPos tempPos = water.mutable();
 
-    /**
-     * Checks if on position "water" really is water, if the water is connected to land and if the pond is big enough (bigger then 20).
-     *
-     * @param world  The world the player is in.
-     * @param water  The coordinate to check.
-     * @param width  which has to be water.
-     * @param length which has to be water.
-     * @return true if water.
-     */
-    public static boolean checkWater(@NotNull final BlockGetter world, @NotNull final BlockPos water, final int width, final int length)
-    {
-        if (world.getBlockState(water).getBlock() != Blocks.WATER || !world.getBlockState(water.above()).isAir())
+        for (int i = 1; i < WATER_DEPTH_REQUIREMENT; i++)
         {
-            return false;
-        }
-
-        final int x = water.getX();
-        final int y = water.getY();
-        final int z = water.getZ();
-
-        //If not one direction contains a pool with length at least 6 and width 7
-        return checkWaterPoolInDirectionXThenZ(world, x, y, z, 1, width, length)
-                 || checkWaterPoolInDirectionXThenZ(world, x, y, z, -1, width, length)
-                 || checkWaterPoolInDirectionZThenX(world, x, y, z, 1, width, length)
-                 || checkWaterPoolInDirectionZThenX(world, x, y, z, -1, width, length);
-    }
-
-    /**
-     * Checks if all blocks in direction X are water and if yes from the middle to both sides in. direction Z all blocks are also water.
-     *
-     * @param world  World.
-     * @param x      posX.
-     * @param y      posY.
-     * @param z      posZ.
-     * @param vector direction.
-     * @param width  which has to be water.
-     * @param length length has to be water.
-     * @return true if all blocks are water, else false.
-     */
-    private static boolean checkWaterPoolInDirectionXThenZ(
-      @NotNull final BlockGetter world,
-      final int x,
-      final int y,
-      final int z,
-      final int vector,
-      final int width,
-      final int length)
-    {
-        //Check 6 blocks in direction +/- x
-        for (int dx = x - (width / 2 * vector); dx <= x + (width / 2 * vector); dx++)
-        {
-            if (world.getBlockState(new BlockPos(dx, y, z)).getBlock() != Blocks.WATER)
+            if (world.getBlockState(tempPos.set(water.getX(), water.getY() - i, water.getZ())).getBlock() != Blocks.WATER)
```

Comment:
This might be problematic, there are some styles that have decorative plants at the bottom of the pond.  Is it intended that this only checks one position?

Also, instead of using `tempPos`, this could just use `water.below(i)`.